### DISCUSSION TITLE
Custom font Turkish character

### DISCUSSION THREAD

**Discussion by yunusemreucan** [Bodmer/TFT_eSPI#2917] [Q&A - Fonts] [answered]
Hello 
I used FontConverter.exe for my different fonts
My font is Swiss721BT.h file 
But I couldn't show Turkish character it on LCD
For example (ı o ğ ü ş ö ç) characters
How I include this font.h file
Thanks

**Comment by Bodmer** [comment]
I am not sure how those characters will be mapped to point (e.g. ASCII or Unicode values) codes. Post the font.h file and an example sketch in a zip file.

**Reply by yunusemreucan** [reply]
Hello, I had sent you a question about the issue of not displaying Turkish characters on the screen about a week ago. I was going to send you the files related to these, but I now have the time. The files are attached.                                                                       https://drive.google.com/file/d/1N5wYFZhypJYfknmtlqmP8DI5tglGS1Qj/view?usp=sharing

**Reply by yunusemreucan** [reply, marked as answer]
I found the solution with OpenFontRender, thank you

**Reply by Bodmer** [reply]
OK, great. Sorry I did not get back earlier.

**Reply by Bodmer** [reply]
I had a look at the font file you posted and it does not contain turkish characters, the byte sequence "0x63, 0x18, 0xC6, 0x31, 0x8F, 0xC0, 0xFC" repeats for each character repeats from code 0x7F to 0x9C (except 0x80 which prints as the Euro symbol). That byte sequence just prints a box.

This indicates that the font generator used did not include the correct characters.

You have made the correct step using OpenFontRender. The Arduino IDE will encode the characters in UTTF8 format and then the library can interpret those characters (point codes).  The only other form format you could use for Turkish characters is the "Smooth font" format or [this support library](https://github.com/Bodmer/U8g2_for_TFT_eSPI).

**Reply by yunusemreucan** [reply]
Thanks your reply.  Best regards
